### ISSUE TITLE
composite primary key

### ISSUE THREAD

**Issue by Simsal** [techfort/LokiJS#450]
Hi guys,

I know that like this i can define my own primary key:
var collection = db.addCollection("test", [
  unique: ["name"]
]);

Is there a possibility to use a composite primary key, something like 
unique: ["name" + "birthday"]
?

Greetings!

**Comment by VladimirTechMan** [comment]
@Simsal, no, the current design of Loki does not support that functionality. Sorry.

**Comment by woyodb** [comment]
Same problem,  And I solved it by catching the 'pre-insert' event to constrains the compound unique key manually.
The code may like this : 
```
my_coll.on('pre-insert', o=>{
	let uk = ['k1','k2','k3']; //compound unique keys
	if(uk && uk.length>1){ 
		let q = {};
		uk.forEach(f=>q[f] = o[f]);
		if(my_coll.findOne(q)) throw new Error(`Duplicate unique key for: ${JSON.stringify(q)}` ); 
	}
});
```
Hope to be useful for someone that have the same question.
